cache.go

```diff
@@ -284,7 +282,10 @@ func (c *Cache[K, V]) Set(key K, value V, ttl time.Duration) *Item[K, V] {
 // expiration timestamp on successful retrieval.
 // If the item is not found, a nil value is returned.
 func (c *Cache[K, V]) Get(key K, opts ...Option[K, V]) *Item[K, V] {
-	return c.getWithOpts(key, opts...)
+	c.items.mu.Lock()
+	defer c.items.mu.Unlock()
```

Comment:
This will still cause the aforementioned problem with the loader. We probably need to add another bool parameter to the `getWithOpts` that would control whether the mutex is used or not when calling `elem := c.get(key, !getOpts.disableTouchOnHit)`. 

You can leave this as is, because this PR is getting a bit long as it is. I'll open another PR after we merge this.